_include/q/_utils.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+################################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+CURRENT_DIR=`cd "$(dirname "$0")" && pwd -P`
```

Comment:
```suggestion
SCRIPT_DIR=$(cd $(dirname "${BASH_SOURCE[0]}") && pwd)
```
* `$0` doesn't work well with `source`. We would be on the save side to use `${BASH_SOURCE[0]}` instead. 
* nit: `$()` might be more future proof than using the legacy backticks
* `$CURRENT_DIR` is ambiguous (it could be also the "current working directory"). `SCRIPT_DIR` sounds more specific here. WDYT?